### PR TITLE
Changes to default.nix to be able to use Dune.

### PR DESCRIPTION
**Pull request by Zimmi48**
**Kind:** infrastructure.

Question in passing: @maximedenes @vbgl should we form a nix-team (like there is a ci-team #8023 and a doc-team #8250) for shared code ownership of the Nix files?

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -45,6 +45,7 @@ stdenv.mkDerivation rec {
   buildInputs = [
     hostname
     python2 time # coq-makefile timing tools
+    jbuilder # why is this not part of ocamlPackages and why isn't it called dune yet?
```
Comment (vbgl) [inline default.nix]: What is the purpose of this comment? Here are answers to the questions.

`jbuilder` is no OCaml library; its behavior does not depend on the version of the OCaml compiler used to compile it; therefore, it does not belong to the `ocamlPackages` set.

It is indeed not renamed yet, probably by lack of manpower.

Comment (Zimmi48) [reply default.nix]: Thanks. It was mostly an poor-man's inline comment. I'll remove it.

**Comment by Zimmi48** [comment]
Note that #8412 is requiring Dune 1.1 so we can either wait to merge this PR until nixpkgs has the updated version, or wait to merge #8412 instead.

**Comment by Zimmi48** [comment]
#8412 is going to get merged, so let's wait for a nixpkgs update before merging this one.

**Comment by ejgallego** [comment]
> #8412 is going to get merged, so let's wait for a nixpkgs update before merging this one.

The NiX update should be trivial @Zimmi48 , Dune 1.1 is fully compatible with 1.0

**Comment by vbgl** [comment]
I just tried: it breaks the `re` tests (at least with `nixpkgs` on darwin)… This is irrelevant for Coq but it shows that this is neither “trivial” nor “fully compatible”.

What can be done, is to override it in our `default.nix`:

```
let dune =
  overrideDerivation jbuilder (o: {
    name = "dune-1.1.1";
    src = fetchFromGitHub {
      owner = "ocaml";
      repo = "dune";
      rev = "1.1.1";
      sha256 = "0v2pnxpmqsvrvidpwxvbsypzhqfdnjs5crjp9y61qi8nyj8d75zw";
    };
  });
in
```

**Comment by Zimmi48** [comment]
Good idea @vbgl. Let's do this.

**Comment by Zimmi48** [comment]
Maybe I should move `dune` to the shell-only part until we actually start using this in `nix-build` as well...

**Comment by ejgallego** [comment]
> it breaks the re tests (at least with nixpkgs on darwin)

Fairly odd, that looks like a Dune bug. So far the update on OPAM only broke a handful of developments and that was due to a bug in Dune 1.1.0 [fixed in 1.1.1] But yeah, that's a young tool.

**Review by vbgl — APPROVED** [review]

**Pull request merged**